gapic/schema/wrappers.py

```diff
@@ -540,6 +543,15 @@ def flattened_fields(self) -> Mapping[str, Field]:
         # Done; return the flattened fields
         return answer
 
+    @utils.cached_property
+    def legacy_flattened_fields(self) -> Mapping[str, Field]:
+        """Return the legacy flattening interface: top level fields only,
+        required fields first"""
+        required, optional = utils.partition(lambda f: f.required,
```

Comment:
Is this _the only_ method generated from the rpc (or, does Python ignore `method_signature` annotation`)?

Also, does this mean changing a field from required to optional may cause breaking change?

Reply:
Python has no built in concept of function overloads. It ignores all `method_signature` annotations.

Yes, a field changing from required to optional may cause a breaking change in the client surface. To the best of my understanding, this is true for the monolith as well and was one of the reasons @lukesneeringer changed the method interface semantics.

Reply:
Okay cool. I asked this mainly because of a thread in the chat (I'll cc you). I think the code LGTM if the goal is to preserve the same behavior the monolithic does. But the potential breaking change may be something worth considering for the next major version.

Reply:
The plan is to transition to the microgenerator by making as few breaking changes as possible. Once the transition is complete we can revisit interface improvements.